apis/metal3.io/v1alpha1/baremetalhost_types.go

```diff
@@ -407,6 +407,10 @@ type BareMetalHostSpec struct {
 	// A custom deploy procedure.
 	// +optional
 	CustomDeploy *CustomDeploy `json:"customDeploy,omitempty"`
+
+	// BMH architecture.
```

Comment:
I'd like to see a little more documentation here.

* 'architecture' = CPU architecture
* allowed values (historically we have used `x86_64`, so I think the allowed values are `x86_64`, `aarch64`, `ppc64le`, &c. (not `x86-64` or golang CPU arch values like `amd64, `arm64`, &c.)
* where this is used - it's passed to the preprovisioningimage

Reply:
Thanks, I have modified @zaneb